src/lib/identity/api.ts

```diff
@@ -3,212 +3,53 @@ import type {
-					id,
-					publicFields: {
-						displayName,
-					},
-					dates: { accountCreatedDate },
-					statusFields: {
-						userEmailValidated,
-					},
-					rawResponse: cookieData,
-				};
-			}
-		}
-	}
-
-	return userFromCookieCache;
-};
-
-/**
- * Fetch the logged in user's Google Tag ID from IDAPI
- * @returns one of:
- * - string - the user's Google Tag ID
- * - null - if the request failed
- */
-const fetchGoogleTagIdFromApi = (): Promise<string | null> =>
-	fetch(`${idApiRoot}/user/me/identifiers`, {
-		mode: 'cors',
-		credentials: 'include',
-	})
-		.then((resp) => {
-			if (resp.status === 200) {
-				return resp.json() as Promise<{ googleTagId: string }>;
-			}
-			throw new Error(resp.status.toString());
-		})
-		.then((json) => json.googleTagId)
-		.catch((e) => {
-			console.log('failed to get Identity user identifiers', e);
-			return null;
-		});
+type AuthStatus = SignedOut | SignedIn;
 
-const isUserLoggedIn = (): boolean => getUserFromCookie() !== null;
+const identityAuth = getIdentityAuth();
 
 const getAuthStatus = async (): Promise<AuthStatus> => {
-	if (useOkta) {
-		try {
-			const { isSignedInWithOktaAuthState } = await import('./okta');
-			const authState = await isSignedInWithOktaAuthState();
-			if (authState.isAuthenticated) {
-				return {
-					kind: 'SignedInWithOkta',
-					accessToken: authState.accessToken,
-					idToken: authState.idToken,
-				};
-			}
-		} catch (e) {
-			console.error(e);
-		}
-
-		return {
-			kind: 'SignedOutWithOkta',
-		};
-	}
-	if (isUserLoggedIn()) {
+	const { isAuthenticated, accessToken, idToken } =
+		await identityAuth.isSignedInWithAuthState();
+	if (isAuthenticated) {
 		return {
-			kind: 'SignedInWithCookies',
+			kind: 'SignedIn',
+			accessToken,
+			idToken,
 		};
 	}
 	return {
-		kind: 'SignedOutWithCookies',
+		kind: 'SignedOut',
 	};
 };
 
-const isUserLoggedInOktaRefactor = (): Promise<boolean> =>
-	getAuthStatus().then((authStatus) =>
-		authStatus.kind === 'SignedInWithCookies' ||
-		authStatus.kind === 'SignedInWithOkta'
-			? true
-			: false,
-	);
-
-/**
- * Decide request options based on an {@link AuthStatus}. Requests to authenticated APIs require different options depending on whether
- * you are in the Okta experiment or not.
- * @param authStatus
- * @returns where `authStatus` is:
- *
- * `SignedInWithCookies`:
- * - set the `credentials` option to `"include"`
- *
- * `SignedInWithOkta`:
- * - set the `Authorization` header with a Bearer Access Token
- * - set the `X-GU-IS-OAUTH` header to `true`
- */
-const getOptionsHeadersWithOkta = (
-	authStatus: SignedInWithCookies | SignedInWithOkta,
-): RequestInit => {
-	if (authStatus.kind === 'SignedInWithCookies') {
-		return {
-			credentials: 'include',
-		};
-	}
-
-	return {
-		headers: {
-			Authorization: `Bearer ${authStatus.accessToken.accessToken}`,
-			'X-GU-IS-OAUTH': 'true',
-		},
-	};
-};
+const isUserLoggedIn = (): Promise<boolean> => getIdentityAuth().isSignedIn();
```

Comment:
I know it's not something you've changed, but does this hit the API every time it's called?